nemo_skills/inference/generate_solutions.py

```diff
@@ -133,29 +134,50 @@ def generate_solutions(cfg: GenerateSolutionsConfig):
 
             data_points.append(data_point)
 
-            if len(data_points) == cfg.batch_size:
-                # batch-computing the outputs
-                outputs = llm.generate(
-                    prompts=[prompt.build_string(data_point) for data_point in data_points],
-                    stop_phrases=list(cfg.prompt.stop_phrases),
-                    **asdict(cfg.inference),
-                )
-
-                for output, original_data_point in zip(outputs, data_points):
-                    # to make it easier to follow up with evaluation and limit accidental errors, we are adding
-                    # all of the ground-truth data to the output file alongside the generated solutions
-                    output.update(original_data_point)
-                    if 'error_message' not in output:
-                        output['error_message'] = extract_error_message(output['generation'])
-
-                    if cfg.generation_key != "generation":
-                        output[cfg.generation_key] = output.pop("generation")
-
-                    fout.write(json.dumps(output) + "\n")
-                data_points = []
+            if len(data_points) == cfg.batch_size or idx == len(data) - 1:
```

Comment:
Doesn't this mean that if use_batch_api=True and then default batch_size=16, then we will submit many batch_generate requests, which will override the same file? Or do I misunderstand something here? If that's the case, probably best to put it out of loop to make sure it's always done with full data and batch_size will be ignored in this case

Reply:
It does, I am not sure if openai has any restriction on the batch size, but for my experiments I was using 1024 so that the whole dataset would be covered in a single batch. 
If openai does not have any restriction, then I can modify the code so that it takes batch size nearest power of 2 that is greater then the actual dataset size

Reply:
Why does it need to be a power of 2? Let's just always send the whole data. If there is some restriction or error, users can always manually split the file. If we find ourselves doing this too often in the future, we can automate that, but I'd suggest to use a simpler approach for now

Reply:
Done